stellarphot/tests/test_core.py

```diff
@@ -9,3 +15,105 @@ def test_camera_attributes():
     assert c.gain == gain
     assert c.dark_current == dark_current
     assert c.read_noise == read_noise
+
+
+def test_base_enhanced_table():
+    # Define a realistic table of photometry data
+    data = np.array([[1, 78.17278712191920, 22.505771480719400, 31.798216414544900,
+                      31.658750534057600, 9.294325523269860, 13.02511260943810,
+                      13.02511260943810, 13.02511260943810]])
+    colnames = ['id', 'RA', 'Dec', 'sky_per_pix_avg', 'sky_per_pix_med', 'sky_per_pix_std',
+                'fwhm_x', 'fwhm_y', 'width']
+    coltypes = ['<i8', '<f8', '<f8', '<f8', '<f8', '<f8', '<f8', '<f8', '<f8']
+    colunits = [None,  u.deg,  u.deg,  u.adu,  u.adu,  u.adu,  None,  None,  None]
+    testdata = Table(data, names=colnames, dtype=coltypes, units=colunits)
+
+    # Create several test descriptions
+    test_descript = np.array([['id', '<i8', None, 'id'],
+                                ['RA', '<f8', u.deg, 'ra'],
+                                ['Dec', '<f8', u.deg, 'dec'],
+                                ['sky_per_pix_avg', '<f8', u.adu, 'spp_avg'],
+                                ['sky_per_pix_med', '<f8', u.adu, 'spp_med'],
+                                ['sky_per_pix_std', '<f8', u.adu, 'spp_std'],
+                                ['fwhm_x', '<f8', None, None],
+                                ['fwhm_y', '<f8', None, None],
+                                ['width', '<f8', None, 'fwhm']])
+
+    # This should create a blank data set
+    test_base = BaseEnhancedTable(test_descript)
+
+    assert type(test_base.data) == Table
+    assert len(test_base.ra) == 0
+    assert len(test_base.dec) == 0
+
+    # Should create a populated dataset properly and display the astropy data
+    test_base2 = BaseEnhancedTable(test_descript, testdata)
+    assert len(test_base2.ra) == 1
+    assert len(test_base2.dec) == 1
+
+    # Should raise exception because no inputs are passed
+    with pytest.raises(Exception):
```

Comment:
We should check the exception type too, though if `table_description` is a required argument we probably don't need to test  whether an exception is raised when the argument isn't given.

I do think it makes sense to check each of the cases where the initialization can fail (with a `TypeError` or `ValueError`)

Reply:
Done.